### PR TITLE
http-client-java, eng, bump npm packages

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
both tcgc and resource-manager got a patch bump, not sure what changed though.

preparing a patch release, before https://github.com/microsoft/typespec/pull/4738/
that patch mostly for bumping the npm dependencies, as these patch may intended to fix some problem (and we don't want them get mixed with the bigger PR). we do not expect sdk code change upon this patch.

### CODE REVIEW

File: packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java [hunk 1]
```diff
@@ -63,10 +65,8 @@ public void testTopLevelTrackedResource() {
             .actionSync(new NotificationDetails().withMessage(NOTIFICATION_DETAILS_MESSAGE).withUrgent(true));
 
         // TopLevelTrackedResources.listByResourceGroup
-        List<TopLevelTrackedResource> topLevelTrackedResourceList = manager.topLevels()
-            .listByResourceGroup(RESOURCE_GROUP_NAME)
-            .stream()
-            .collect(Collectors.toList());
+        List<TopLevelTrackedResource> topLevelTrackedResourceList
+            = manager.topLevels().listByResourceGroup(RESOURCE_GROUP_NAME).stream().collect(Collectors.toList());
```
Comment (XiaofeiCao) [inline packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java:69]: What's the cause of the format change?

**Review by XiaofeiCao — APPROVED** [review]

File: packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java:69]: I think via Alan's spotless.

`mvn package` would trigger the formatting.

File: packages/http-client-java/generator/http-client-generator-test/tsp/nested-enum-discriminator.tsp
Comment (weidongxu-microsoft) [inline packages/http-client-java/generator/http-client-generator-test/tsp/nested-enum-discriminator.tsp:1]: remove cadl-ranch-expect from this file

File: packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java (hunk 1 above)
Comment (XiaofeiCao) [reply packages/http-client-java/generator/http-client-generator-test/src/test/java/com/azure/resourcemanager/resources/ArmResourceTest.java:69]: I see.. I didn't include `-Ptest` in `mvn package`.

**Pull request merged**